module/extension/Director/src/Director.cpp

```diff
@@ -215,43 +229,89 @@ namespace module::extension {
 
         // A task has arrived, either it's a root task so we send it off immediately, or we build up our pack for when
         // the Provider has finished executing
-        on<Trigger<BehaviourTask>, Sync<BehaviourTask>>().then(
-            "Director Task",
-            [this](const std::shared_ptr<const BehaviourTask>& task) {
-                // Root level task, make the pack immediately and send it off to be executed as a root task
-                if (providers.count(task->requester_id) == 0) {
-                    emit(std::make_unique<TaskPack>(TaskPack({task})));
+        on<Trigger<BehaviourTask>, Sync<Director>>().then("Director Task", [this](const BehaviourTask& t) {
+            // Make our own mutable copy of the task
+            auto task = std::make_shared<BehaviourTask>(t);
```

Comment:
something about this being a shared_ptr kind of feels awkward to me. two (probably oversimplistic, naive) things I think are:
- this is shared because the pack_builder maps to shared_ptr<behaviourtask>. Could the pack_builder own these? I think it could, because they're being transferred to being owned by the pack [here](https://github.com/NUbots/NUbots/blob/e834d2807364a2962d0bde8baae28ad93323fb9d/module/extension/Director/src/Director.cpp#L304). So I am thinking the pack_builder could map to unique_ptrs perhaps
- We're deref'ing the pointer a bunch here - would it be more efficient to create it on the stack then move/copy it to the heap when you add it to the pack_builder map? (I honestly don't know this one - just curious whether you think the derefs or copy/move would be more expensive)

I feel like this kind of unimportant, surface level stuff is the kind of stuff I'll be able to provide for this review haha

Reply:
It might be able to be made into a unique pointer, but it would influence a lot of places that would have to become raw pointers which I'm not a fan of... (e.g. watchers)

Reply:
Raw pointers, or, references to the unique ptr's data? same same, but different - stealing the idea of using refs to unique ptr data from you 😅

Reply:
Raw pointers I think, or std::ref objects which I'm not a fan of... Because you can't have an `std::vector` of references

Reply:
As for your comment about dereferencing lots of dereferencing vs one is not really any more expensive. Any half decent optimiser will just do it once to get the memory and work with that memory from there

Reply:
Wouldn't even matter if it weren't created here, since it's not declared as volatile the compiler can just assume the value stays the same the entire time. It actually can cause a class of bug when multithreading